proposals/008-topic-name-lookup-facility.md

```diff
@@ -0,0 +1,104 @@
+# Proposal 008: Resolve Topic Names from Topic IDs in the Filter Framework
+
+## Summary
+
+This proposal outlines a new feature for the Kroxylicious filter framework: a mechanism to resolve topic names from their corresponding topic IDs. This will enable filters to implement business logic based on human-readable topic names, rather than the underlying Kafka implementation detail of topic IDs.
+
+---
+
+## Current Situation
+
+Currently, the Kroxylicious framework does not provide a direct way for filters to translate a topic ID back into its topic name. Filters that receive requests containing only topic IDs have no simple, framework-supported method to determine the names of the topics they are interacting with.
+
+---
+
+## Motivation
+
+Apache Kafka increasingly uses **Topic IDs** as unique and stable identifiers for topics, especially in newer RPCs. This approach robustly handles edge cases, such as when a topic is deleted and later recreated with the same name. You can find more details in [KIP-516: Topic Identifiers](https://cwiki.apache.org/confluence/display/KAFKA/KIP-516%3A+Topic+Identifiers). In Kafka 4.1.0 Produce Requests will begin using
+topic ids, impacting multiple Filters offered by the project.
+
+However, humans and business rules operate on **topic names**, not opaque IDs. For example, a policy is typically defined as "Encrypt data for the `sensitive_data` topic" or "Validate schemas for the `invoices` topic." Kroxylicious filters, which enforce these policies, are naturally designed to work with topic names.
+
+To bridge this gap and allow filters to operate effectively, the framework must provide a reliable way to convert topic IDs back to their corresponding names.
+
+---
+
+## Proposal
+
+I propose:
+1. adding a new method to the `FilterContext` interface that allows filters to look up topic names for a given set of topic IDs.
+2. This will drive a request to the upstream to learn the topic names. 
+3. We will cache the topic names at the edge of the proxy/upstream per channel.
+4. Existing Filters that mutate MetadataResponse will compose with this API without modification.
+
+### 1. API Addition
+
+The following method will be added to the `FilterContext`:
```

Comment:
It seems odd to me that you can go from topic uuid to name but not the other way round.  Is there something that makes you confident that the queries will always go that way?